src/app/event/[id]/EventDetail.tsx

```diff
@@ -1,18 +1,103 @@
-import { getEvent } from '../../../api/events';
+'use client';
+
+import format from 'date-fns/format';
+import cx from 'classnames';
+import { useState } from 'react';
+
+import { HeartIcon } from '../../../components/icons/HeartIcon';
+import { mockEvent } from './mock';
+import { Event } from '../../../types/Event';
 
 type EventDetailProps = {
-  id: string;
+  event: Event;
+  className?: string;
 };
 
-export const EventDetail: React.FC<EventDetailProps> = async ({ id }) => {
-  const event = await getEvent(id);
+export const EventDetail: React.FC<EventDetailProps> = ({
+  event,
+  className = '',
+}) => {
+  const [isLiked, setIsLiked] = useState(false);
+
+  const handleLikeEvent = () => {
+    setIsLiked(!isLiked);
+  };
+
+  const { sport, location, price, date, level } = event;
 
   return (
-    <>
-      <h1 className="mt-24 px-20 text-center text-2xl font-medium leading-normal md:mt-14 md:px-0 lg:text-start lg:text-4xl">
-        {event?.name}
-      </h1>
-      <p className="mt-8">{event?.description}</p>
-    </>
+    <section
+      className={`${className} top-14 w-full max-w-lg rounded-md bg-white p-7 text-lg shadow-md lg:max-w-md`}
```

Comment:
use `cx` and you wont have to default the prop. I recommend rather learning to wrap the className in cx than trying to remember defaulting of the `className` in the prop destructuring